protocol/blockfetch/client.go

```diff
@@ -196,6 +200,8 @@ func (c *Client) messageHandler(msg protocol.Message) error {
 		err = c.handleBlock(msg)
 	case MessageTypeBatchDone:
 		err = c.handleBatchDone()
+	case MessageTypeClientDone:
+		return nil
```

Comment:
This doesn't belong here. The client will never receive the `ClientDone` message, only the server.